Sources/Foundation/Process.swift

```diff
@@ -950,6 +957,24 @@ open class Process: NSObject, @unchecked Sendable {
         }
 #endif
 
+        // Unsafe fallback for systems missing posix_spawn_file_actions_addchdir[_np]
+        // This includes Glibc versions older than 2.29 such as on Amazon Linux 2
+        if useFallbackChdir {
+            let fileManager = FileManager()
+            let previousDirectoryPath = fileManager.currentDirectoryPath
+            if let dir = currentDirectoryURL?.path, !fileManager.changeCurrentDirectoryPath(dir) {
+                throw _NSErrorWithErrno(errno, reading: true, url: currentDirectoryURL)
+            }
+        }
+
+        defer {
+            if useFallbackChdir {
+                // Reset the previous working directory path.
+                let fileManager = FileManager()
+                _ = fileManager.changeCurrentDirectoryPath(previousDirectoryPath)
```

Comment:
This `previousDirectoryPath` is out of scope right?

Reply:
Whoops, fixed